core/trino-main/src/main/java/io/trino/sql/planner/optimizations/PropertyDerivations.java

```diff
@@ -661,10 +659,11 @@ public ActualProperties visitExchange(ExchangeNode node, List<ActualProperties>
                     if (inputProperty.isEffectivelySingleStream() && node.getOrderingScheme().isEmpty()) {
```

Comment:
I think `isEffectivelySingleStream()` might be incorrectly returning `true` here.